app/services/resource_service.rb

```diff
@@ -42,6 +42,7 @@ def create_or_recreate(old_record, external_resource)
       external_resource.destroy if external_resource.present?
       if @resource.save
         create_external_resource
+        FacebookService.delay.send_post_to_page(resource_id: @resource.id, resource_type: @resource_class.name)
```

Comment:
glaube wäre besser zu lesen, wenn hier die prüfung wäre vor "post to page", ob es überhaupt geschehen soll und nicht innerhalb der "post to page"-methode.
der resource service sollte die settings der ressources prüfen denk ich und nicht die methoden die dann aufgerufen werden?

Reply:
ich weiß nicht ob es hierfür eine schlaue Antwort/Lösung/Paradigma gibt. 
Aktuell ist die `send_post_to_page` in sich geschlossen und kann überall wieder aufgerufen werden ohne vorab Prüfungen machen zu müssen.
So kann es auch nicht passieren, dass Daten an Facebook gesendet werden, die per Definition in den Settings nicht dahin gesendet werden sollen.

Andererseits "passiert" derzeit "zu viel" in der Methode. Wenn man dort aber Sachen rauszieht, würde zu viel in der Methode `create_or_recreate` passieren. Dann wäre die Schlussfolgerung da noch eine Serviceklasse drum herum zu bauen.

Ich erstelle hierzu mal ein Issue für eine Verbesserungsmöglichkeit